tests/tasks/test_perturbation_expression_prediction.py

```diff
@@ -259,88 +225,79 @@ def test_perturbation_expression_prediction_task_wilcoxon():
     assert np.allclose(task_output.true_log_fc_dict["gene_A"], true_lfc_gene_A)
     assert np.allclose(task_output.true_log_fc_dict["gene_B"], true_lfc_gene_B)
 
-    # Then, run the full task and validate metrics are perfect
+    # Test full task execution
     results = task.run(cell_representation, task_input)
+    assert_metric_results(
+        results,
+        expected_count=2,
+        expected_types={MetricType.SPEARMAN_CORRELATION_CALCULATION},
+        perfect_correlation=True,
+    )
 
-    assert isinstance(results, list)
-    assert all(isinstance(r, MetricResult) for r in results)
-    # Task now returns one metric per condition
-    assert len(results) == 2
-
-    # Each result should have perfect correlation
-    for r in results:
-        assert np.isclose(r.value, 1.0)
-
-    # Verify metric types present
-    metric_types = {result.metric_type for result in results}
-    expected_types = {MetricType.SPEARMAN_CORRELATION_CALCULATION}
-    assert expected_types.issubset(metric_types)
 
+def test_task_uses_strict_pair_mapping_for_pred_lfc(simple_toy_data, adata_uns_setup):
+    """Test that task uses strict pair mapping for prediction log fold changes."""
+    adata = simple_toy_data["adata"]
+    de_results = simple_toy_data["de_results"]
+    predictions = simple_toy_data["predictions"]
 
-def test_task_uses_strict_pair_mapping_for_pred_lfc():
-    # Build toy dataset: 2 genes, 4 cells (2 control, 2 treated for pertA)
-    X = np.array(
-        [
-            [1.0, 2.0],  # NT1
-            [2.0, 1.0],  # NT2
-            [3.0, 4.0],  # T1
-            [5.0, 3.0],  # T2
-        ]
+    # Add required UNS data using helper
+    adata_uns_setup(
+        adata,
+        de_results,
+        {"pertA": ["g1", "g2"]},
+        {"pertA": {"T1": "NT1", "T2": "NT2"}},
     )
-    obs = pd.DataFrame(
-        {
-            "condition": ["ctrl", "ctrl", "pertA", "pertA"],
-        },
-        index=["NT1", "NT2", "T1", "T2"],
-    )
-    var = pd.DataFrame(index=["g1", "g2"])
-    adata = ad.AnnData(X=X, obs=obs, var=var)
 
-    # Provide required UNS for the task
-    de = pd.DataFrame(
-        {
-            "condition": ["pertA", "pertA"],
-            "gene_id": ["g1", "g2"],
-            "logfoldchange": [1.0, 2.0],
-            "pval_adj": [1e-6, 1e-4],
-        }
-    )
-    adata.uns["de_results"] = de
-    adata.uns["metric_column"] = "logfoldchange"
-    adata.uns["control_cells_map"] = {"pertA": {"T1": "NT1", "T2": "NT2"}}
-    adata.uns["target_conditions_dict"] = {"pertA": ["g1", "g2"]}
-
-    # Use log1p as prediction matrix to satisfy log-normalization check
-    preds = np.log1p(X)
-    task = PerturbationExpressionPredictionTask(
-        condition_key="condition",
-        control_name="ctrl",
-    )
     task_input = build_task_input_from_predictions(
         predictions_adata=adata, dataset_adata=adata
     )
+    task = PerturbationExpressionPredictionTask()
+    output = task._run_task(predictions, task_input)
```

Comment:
Private methods in child classes (task._run_task) should not be used instead of the public method in the parent class (task.run).